api/api/docs/base_docs.py

```diff
@@ -83,3 +98,123 @@ def get_operation_id(self) -> str:
             else:
                 operation_tokens.append("detail")
         return "_".join(operation_tokens)
+
+
+source_404_message = "Invalid source 'name'. Valid sources are ..."
+source_404_response = OpenApiResponse(
+    NotFoundErrorSerializer,
+    examples=[
+        OpenApiExample(
+            name="404",
+            value={"detail": source_404_message},
+        )
+    ],
+)
+
+
+def build_source_path_parameter(media_type: MediaType):
+    valid_description = (
+        f"Valid values are source_names from the stats endpoint: "
+        f"https://api.openverse.engineering/v1/{media_type}/stats/."
+    )
+
+    return OpenApiParameter(
+        name="source",
+        type=str,
+        location=OpenApiParameter.PATH,
+        description=f"The source of {media_type}. {valid_description}",
+    )
+
+
+creator_path_parameter = OpenApiParameter(
+    name="creator",
+    type=str,
+    location=OpenApiParameter.PATH,
+    description="The name of the media creator. This parameter "
+    "is case-sensitive, and matches exactly.",
+)
+tag_path_parameter = OpenApiParameter(
+    name="tag",
+    type=str,
+    location=OpenApiParameter.PATH,
+    description="The tag of the media. Not case-sensitive.",
+)
+
+
+def get_collection_description(media_type, collection):
+    if collection == "tag":
+        return f"""
+Get a collection of {media_type} with a specific tag.
+
+This endpoint returns only the exact matches. To search within the
+tag values, or to match several tags, use the `search` endpoint
+with `tags` query parameter instead of `q` parameter.
+
+The returned results are ordered primarily based on their popularity
+and authority. However, note that the exact order may vary over time
+or across requests.
+    """
+    elif collection == "source":
+        return f"""
+Get a collection of {media_type} from a specific source.
+
+This endpoint returns only the exact matches. To search within the source value,
+use the `search` endpoint with `source` query parameter instead of `q`.
+
+The results in the collection will be sorted by the order in which they
+were added to Openverse.
+    """
+    elif collection == "creator":
+        return f"""
+Get a collection of {media_type} by a specific creator from the specified source.
+
+This endpoint returns only the exact matches both on the creator and the source.
+Notice that a single creator's media items can be found on several sources, but
```

Comment:
Not only because source =/= provider.
Several sources (and providers) can have works by the same provider. For example, if you search for Rembradt, you can see media from several sources, such as Wikimedia, Met, inaturalist, Finnish Heritage Agency. Not all of them refer to the same creator. Flickr and Inaturalist, for instance, can have users that use "Rembrandt" as their username. This URL allows you to see media by creators with "Rembrandt" in their name, with the sources I listed above excluded: https://api.openverse.engineering/v1/images/?creator=rembrandt&excluded_source=wikimedia,met,inaturalist,finnish_heritage_agency 
You can also see that the same creator can have a different name in different providers (it's "Rembrandt van Rijn" in SMK)